table/tables/partition_test.go

```diff
@@ -530,3 +536,51 @@ func (ts *testSuite) TestIssue21574(c *C) {
 	tk.MustExec("drop table t_21574")
 	tk.MustExec("create table t_21574 (`key` int, `table` int) partition by list columns (`key`,`table`) (partition p0 values in ((1,1)));")
 }
+
+func (ts *testSuite) TestListPartitionRangePrune(c *C) {
+	tk := testkit.NewTestKitWithInit(c, ts.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop tables if exists t1")
+	tk.MustExec("set @@session.tidb_enable_list_partition = ON")
+	tk.MustExec(`create table t1 (id int, a int, b int) partition by list ( a ) (partition p0 values in (1,2,3,4,5), partition p1 values in (6,7,8,9,10), partition p2 values in (null));`)
+	tb, err := ts.dom.InfoSchema().TableByName(model.NewCIStr("test"), model.NewCIStr("t1"))
+	c.Assert(err, IsNil)
+	partExpr, err := tb.(partitionTable).PartitionExpr()
+	c.Assert(err, IsNil)
+	cases := []struct {
+		r      ranger.Range
+		result []int
+	}{
+		{
+			r:      ranger.Range{LowVal: types.MakeDatums(0), HighVal: types.MakeDatums(1)},
+			result: []int{0},
+		},
+		{
+			r:      ranger.Range{LowVal: types.MakeDatums(0), HighVal: types.MakeDatums(6)},
+			result: []int{0, 0, 0, 0, 0, 1}, // need deduplicate the result, the expect should be {0,1}
```

Comment:
As the comment said, the expected result should be `{0,1}`

Reply:
Do you mean this test should fail? And the correct result should be '{0,1}' and Maybe something wrong now ?